# 16s_diadema_ajmc2020
 
This contains all of the data used in the study AJMC 2020.
including qiime commands, etc.

## File Tree Directory

```
 |-- diadema_ajmc2020
    |-- 16S
        |-- RawSequences
            |-- AllSamples
            |-- CategSize
        |-- qiimecommands.sh
    |-- CytoB
        |-- Geneious
            |-- de novo
                |-- Alignments
                    |-- Geneious
                    |-- Velvet
                |-- ConsensusSequences
                |-- Geneious de novo
                    |-- Edited
                    |-- Raw
                    |-- Tree
                |-- Velvet de novo
                    |-- Edited
                    |-- Raw
                    |-- Tree
                |-- Trees
            |-- Map To References
                |-- Alignments
                |-- alignref_Contigs
                |-- ConsensusSeq
            |-- Normalized_ErrorCorrected_RemovedDuplicates
            |-- Trees
        |-- RawSequences
        |-- ReferenceSequences
        |-- Trees
    |-- README.md
    |-- References Thesis.xlx
```

______________________________________________

----------------------------------------------

----------------------------------------------


## WORKFLOW

#### CytoB - Geneious & Mega X

Learn how to use Geneious: https://www.geneious.com/ | 
https://www.geneious.com/tutorials/ | 

Tutorials used in this study:
https://www.geneious.com/tutorials/de-novo-assembly/ | 
https://www.geneious.com/tutorials/map-to-reference/ | 

Learn how to use Mega X:
https://megasoftware.net/ | 

References:

https://bsapubs.onlinelibrary.wiley.com/doi/epdf/10.3732/apps.1400062 | 


1. Download & Import sequences from diadema_ajmc2020\CytoB\RawSequences into Geneious.

2. Merge Pair Ends, Trim, Normalize, Error Correct Sequences, Remove Duplicates. [Remove Chimeric Reads???]

```  
 |-- diadema_ajmc2020
    |-- CytoB
        |-- Geneious
            |-- Normalized_ErrorCorrected_RemovedDuplicates
```
            
3. Velvet de novo Assembly of each of the samples

```
 |-- diadema_ajmc2020
    |-- CytoB
        |-- Geneious
            |-- de novo
                |-- Velvet de novo
                    |-- Raw
```

4. Edits:
The contigs generated by the de novo were edited to only contain sequences which corresponded close to the length of the gene

```
 |-- diadema_ajmc2020
    |-- CytoB
        |-- Geneious
            |-- de novo
                |-- Velvet de novo
                    |-- Edited
```

5. Alignments: 
The edited Contigs were aligned within each sample using Geneious Alignment: Global 51

```
 |-- diadema_ajmc2020
    |-- CytoB
        |-- Geneious
            |-- de novo
                |-- Alignments
                    |-- Velvet
```

7. Consensus Sequences:
A Consensus Sequence was generated using 0 Majority

```
 |-- diadema_ajmc2020
    |-- CytoB
        |-- Geneious
            |-- de novo
                |-- ConsensusSequences
                
```

8. Consensus Trees
A tree was generated with all the samples.

```
 |-- diadema_ajmc2020
    |-- CytoB
        |-- Geneious
            |-- de novo
                |-- Velvet de novo
                    |-- Tree
                    
```
                    


______________________________________________

----------------------------------------------

----------------------------------------------


#### 16S rRNA - Qiime2




Tutorial used:

https://docs.qiime2.org/2020.6/     |       

https://docs.qiime2.org/2020.6/tutorials/moving-pictures/       |       

https://s3-us-west-2.amazonaws.com/qiime2-data/distro/core/virtualbox-images.txt        |       

To view Qiime files (artifacts (.qza files) and visualizations (.qzv files)) without downloading qiime use: https://view.qiime2.org/



References:
1. Bolyen, Evan, et al. “Reproducible, Interactive, Scalable and Extensible Microbiome Data Science Using QIIME 2.” Nature Biotechnology, vol. 37, no. 8, 2019, pp. 852–857., doi:10.1038/s41587-019-0209-9.

2. Estaki, Mehrbod, et al. “QIIME 2 Enables Comprehensive End‐to‐End Analysis of Diverse Microbiome Data and Comparative Studies with Publicly Available Data.” Current Protocols in Bioinformatics, vol. 70, no. 1, 2020, doi:10.1002/cpbi.100.

3. Hall, Michael, and Robert G. Beiko. “16S RRNA Gene Analysis with QIIME2.” Methods in Molecular Biology Microbiome Analysis, 2018, pp. 113–129., doi:10.1007/978-1-4939-8728-3_8.



Workflow:

QIIME2 Commands used can be found in diadema_ajmc2020\16S\qiimecommands.sh


1. Import Data

2. Visualize Data
Files Generated
```
allsamplesimport.qzv
```

3. Filter
```
allsamplesimportdemux-filtered.qza
allsamplesimportdemux-filter-stats.qza
```

4. Deblur & Generate Feature Table
```
allsamplesimportrep-seqs-deblur.qza
allsamplesimporttable-deblur.qza
allsamplesimportdeblur-stats.qza
```

5. Summary Stats
```
allsamplesimportdemux-filter-stats.qzv
allsamplesimportdeblur-stats.qzv
```

6. Summary Feature Table $ Feature Sequence Data
```
seaurchinfeaturetableallsamples.qzv
sequences.qzv
```

7. Generate a Phylogenetic Tree
```
aligned-rep-seqs.qza
masked-aligned-rep-seqs.qza
unrooted-tree.qza
rooted-tree.qza
```

8. Alpha & Beta Diversity
```
core-metrics-results (FOLDER)
```

9. Exploring Microbial Composition with Metadata
```
core-metrics-results/faith-pd-group-significance.qzv
core-metrics-results/evenness-group-significance.qzv

```

10. Permanova
```
core-metrics-results/unweighted-unifrac-reefhabitat-significance.qzv
core-metrics-results/unweighted-unifrac-location-significance.qzv
```

11. Emperor Plots by Metadata
```
core-metrics-results/unweighted-unifrac-emperor-size.qzv
core-metrics-results/bray-curtis-emperor-size.qzv
```

12. Alpha Rarefaction Plotting
```
alpha-rarefaction.qzv
```

13. Taxonomic Analysis
```
taxonomy.qza
taxonomy.qzv
taxa-bar-plots.qzv
```

14. Group by Metadata
```
locationseaurchintable.qza
```

15. Merge by Metadata.
```
grouped-taxa-bar-plots.qzv
```

    
______________________________________________

----------------------------------------------

----------------------------------------------